src/renderers/dom/shared/__tests__/ReactBrowserEventEmitter-test.js

```diff
@@ -435,4 +435,54 @@ describe('ReactBrowserEventEmitter', () => {
     expect(idCallOrder[1] === PARENT).toBe(true);
     expect(idCallOrder[2] === GRANDPARENT).toBe(true);
   });
+
+  describe('local listener attachment', function() {
+    it('does not attach a local listener a second time on update', () => {
+      var container = document.createElement('div');
+      var spy = jest.fn();
+
+      class TestComponent extends React.Component {
+        render() {
+          return <div onTouchMove={spy}>Test</div>;
+        }
+      }
+
+      var component = ReactDOM.render(<TestComponent />, container);
+      var el = container.querySelector('div');
+
+      // Force an update to cause a re-render
+      component.forceUpdate();
```

Comment:
👍. Good catch. I'll update this test.

If I remember, the old listener is definitely getting called. I need to work on that next.